src/OpenTelemetry.Api/Trace/SpanAttributes.cs

```diff
@@ -0,0 +1,118 @@
+﻿// <copyright file="SpanAttributes.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// A class that represents the span attributes.
+    /// </summary>
+    public struct SpanAttributes
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SpanAttributes"/> struct.
+        /// </summary>
+        /// <param name="initialAttributes">Initial attributes for the <see cref="SpanAttributes"/>.</param>
+        public SpanAttributes(IEnumerable<KeyValuePair<string, object>> initialAttributes)
+        {
+            this.Attributes = initialAttributes == null ? new ActivityTagsCollection() : new ActivityTagsCollection(initialAttributes);
+        }
+
+        internal ActivityTagsCollection Attributes { get; }
+
+        /// <summary>
+        /// Add entry to the attributes.
+        /// </summary>
+        /// <param name="key">Entry key.</param>
+        /// <param name="value">Entry value.</param>
+        public void Add(string key, long value)
+        {
+            this.Attributes.Add(key, value);
```

Comment:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-attributes

To striclty follow spec, we need to use this.Attributes[key] = value.

Spec says if same key is used, overwrite it. Lets follow it.

Reply:
should i add a null key check?

Reply:
just created a private function to handle that.